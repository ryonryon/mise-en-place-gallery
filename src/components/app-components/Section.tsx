import styled from "@emotion/styled";
import * as React from "react";

import StoryBookIcon from "../../assets/storybook-icon.svg";

export const StyledSection = styled.section`
  width: 85%;
  margin: 0 auto 100px auto;

  h1 {
    padding-top: 5%;
    margin: -5% 0 5% 0;

    & > a {
      height: 24px;
      width: 24px;
      margin-right: 8px;

      & > img {
        height: 24px;
        width: 24px;
        vertical-align: text-top;
      }
    }
  }
`;

interface Props {
  title: string;
  children: React.ReactNode;
}

const Section = ({ title, children }: Props) => (
  <StyledSection>
    <h1 id={title.toLowerCase()}>
      <a href={`https://mise-en-place-ui-components.herokuapp.com/story?path=/story/${title.toLowerCase()}`}>
        <img src={StoryBookIcon} alt="story-icon" />
      </a>
      {title}
    </h1>
    {children}
  </StyledSection>
);

export default Section;
