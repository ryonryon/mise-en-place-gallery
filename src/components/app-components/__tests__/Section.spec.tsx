import * as React from "react";
import * as renderer from "react-test-renderer";

import Section from "../Section";

describe("Section test", () => {
  it("tests Section renders correctly ", () => {
    const section = renderer.create(<Section title="test">test</Section>).toJSON();
    expect(section).toMatchSnapshot();
  });
});
