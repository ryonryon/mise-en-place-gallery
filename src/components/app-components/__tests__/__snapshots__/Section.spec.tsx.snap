// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section test tests Section renders correctly  1`] = `
<section
  className="css-150x4a8"
>
  <h1
    id="test"
  >
    <a
      href="https://mise-en-place-ui-components.herokuapp.com/story?path=/story/test"
    >
      <img
        alt="story-icon"
      />
    </a>
    test
  </h1>
  test
</section>
`;
